### ISSUE TITLE
Card index does not update when state is updated

### ISSUE THREAD

**Issue by shihaotan91** [alexbrillant/react-native-deck-swiper#30]
I have a constructor that creates a cardIndex state of 0 by default.

```
 constructor(props) {
    super(props);
    this.state = {
      confirm: false,
      cardIndex: 0
    }
  }
```

I have another function that changes the state.

```  
changeView = () => {
    this.setState({
      confirm: true
      cardIndex: 1
    })
    this.swiper.jumpToCardIndex(this.state.cardIndex)
  }

Terms.setStateCallBack(this.changeView);
```

This function is called from another component using a callback.

```
let callbackSetState;

Terms.setStateCallBack = (callback) => {
  callbackSetState = callback;
}

Terms.actions =  <View style={styles.container}>
                    <Button style={styles.text} containerStyle={styles.redeemButton} onPress = {() => callbackSetState()} >
                      Redeem now
                    </Button>
                  </View>
```
And in my swiper component I set the cardIndex to `this.state.cardIndex`

```
<Swiper style={styles.swiper}
            cards={[About, Terms]}
            cardIndex = {this.state.cardIndex}
```

However even when the state is updated when I click on the redeem button, the swiper cardIndex always render 0 first. In other words, I will have to swipe right again to be able to click on the confirm/cancel button. The expected behaviour should be that after the redeem button is pressed, the swiper should automatically render with cardIndex 1 and the confirm/cancel button. 

Using console.log(this.state.cardIndex), I know that the state is changing correctly. But cardIndex on swiper isn't rendering the correct card.

Can someone help me?

These are my cards so it's easier for you to visualise.

Card with index of 0
<img width="380" alt="screen shot 2017-07-06 at 12 49 44 pm" src="https://user-images.githubusercontent.com/22129468/27896178-d99d51b0-624a-11e7-9204-d305f784a19c.png">

Card with index of 1 and confirm: false
<img width="376" alt="screen shot 2017-07-06 at 12 50 02 pm" src="https://user-images.githubusercontent.com/22129468/27896184-e404a64e-624a-11e7-8c6f-c955f675357a.png">

Card with index of 1 with confirm: true
<img width="381" alt="screen shot 2017-07-06 at 12 50 11 pm" src="https://user-images.githubusercontent.com/22129468/27896187-e9337294-624a-11e7-8a77-09f0c8a884d6.png">

**Comment by alexbrillant** [comment]
Try this : 
```

changeView = () => {
    this.setState({
      confirm: true
      cardIndex: 1
    }, () => {
      this.swiper.jumpToCardIndex(this.state.cardIndex)
    })
  }
```

Terms.setStateCallBack(this.changeView);

**Comment by alexbrillant** [comment]
fixed in 1.3.3 !
